fmriprep/interfaces/cifti.py

```diff
@@ -0,0 +1,219 @@
+    'CIFTI_STRUCTURE_BRAIN_STEM': [16],
+    'CIFTI_STRUCTURE_CAUDATE_LEFT': [11],
+    'CIFTI_STRUCTURE_CAUDATE_RIGHT': [50],
+    'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_LEFT': [7],
+    'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_RIGHT': [46],
+    # 'CIFTI_STRUCTURE_CEREBELLUM': [],
+    'CIFTI_STRUCTURE_CEREBELLUM_LEFT': [6],
+    'CIFTI_STRUCTURE_CEREBELLUM_RIGHT': [45],
+    'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_LEFT': [28],
+    'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_RIGHT': [60],
+    'CIFTI_STRUCTURE_HIPPOCAMPUS_LEFT': [17],
+    'CIFTI_STRUCTURE_HIPPOCAMPUS_RIGHT': [53],
+    # 'CIFTI_STRUCTURE_OTHER': [],
+    # 'CIFTI_STRUCTURE_OTHER_GREY_MATTER': [],
+    # 'CIFTI_STRUCTURE_OTHER_WHITE_MATTER': [],
+    'CIFTI_STRUCTURE_PALLIDUM_LEFT': [13],
+    'CIFTI_STRUCTURE_PALLIDUM_RIGHT': [52],
+    'CIFTI_STRUCTURE_PUTAMEN_LEFT': [12],
+    'CIFTI_STRUCTURE_PUTAMEN_RIGHT': [51],
+    'CIFTI_STRUCTURE_THALAMUS_LEFT': [10],
+    'CIFTI_STRUCTURE_THALAMUS_RIGHT': [49],
+}
+
+
+class GenerateCiftiInputSpec(BaseInterfaceInputSpec):
+    bold_file = File(mandatory=True, exists=True, desc="input BOLD file")
+    volume_target = traits.Enum("MNI152NLin2009cAsym", mandatory=True, usedefault=True,
+                                desc="CIFTI volumetric output space")
+    surface_target = traits.Enum("fsaverage5", "fsaverage6", "fsaverage", mandatory=True,
+                                 usedefault=True, desc="CIFTI surface target space")
+    subjects_dir = Directory(mandatory=True, desc="FreeSurfer SUBJECTS_DIR")
+    TR = traits.Float(mandatory=True, desc="repetition time")
+    gifti_files = traits.List(File(exists=True), mandatory=True,
+                              desc="surface geometry files")
+
+
+class GenerateCiftiOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc="generated CIFTI file")
+
+
+class GenerateCifti(SimpleInterface):
+    """
+    Generate CIFTI image from BOLD file in target space
+    """
+    input_spec = GenerateCiftiInputSpec
+    output_spec = GenerateCiftiOutputSpec
+
+    def _run_interface(self, runtime):
+        self.annotation_files, self.label_file = self._fetch_data()
+        self._results["out_file"] = create_cifti_image(
+            self.bold_file,
+            self.label_file,
+            self.annotation_files,
+            self.gifti_files,
+            self.volume_target,
+            self.surface_target,
+            self.TR)
+
+        return runtime
+
+    def _fetch_data(self):
+        """Converts inputspec to files"""
+        if self.surface_target == "fsnative":
+            raise NotImplementedError
+
+        annotation_files = sorted(glob(os.path.join(self.subjects_dir,
+                                                    self.surface_target,
+                                                    'label',
+                                                    '*h.aparc.annot')))
+        # TODO: fetch label_file
+        return annotation_files
```

Comment:
Does it make sense to use a `FreeSurferSource` to find the `aparc.annot` files?

Also, do we want to consider the possibility people will want to use DKTatlas or a2009s (at least to the extent of being aware of whether we're making a move in that direction easy or hard)?

Reply:
We could but we'd still have to parse the results - I can change if you feel strongly.

Could you elaborate on your second point? Use how?

Reply:
FreeSurfer now automatically produces

```Shell
% ls /data/out/ds000114/derivatives/freesurfer/sub-07/label/lh.aparc*.annot 
/data/out/ds000114/derivatives/freesurfer/sub-07/label/lh.aparc.a2009s.annot
/data/out/ds000114/derivatives/freesurfer/sub-07/label/lh.aparc.annot
/data/out/ds000114/derivatives/freesurfer/sub-07/label/lh.aparc.DKTatlas.annot
```

It seems plausible that someone might want to generate their CIFTI files with labels from another parcellation. Again, not necessarily that we need to code it up right now, but just to consider how easy it would be to augment.

(It may also be that I'm horribly wrong and CIFTI pretty much demands vanilla aparc.)